tsdb/compact.go

```diff
@@ -778,6 +779,25 @@ func AllSortedPostings(ctx context.Context, reader IndexReader) index.Postings {
 
 type DefaultBlockPopulator struct{}
 
+// countBuckets is a helper function that counts buckets given lengths and zero count.
+func countBuckets(posLen, negLen int, hasZero bool) uint64 {
+	count := uint64(posLen + negLen)
+	if hasZero {
+		count++
+	}
+	return count
+}
+
+// countHistogramBuckets returns the number of buckets in a histogram.
+func countHistogramBuckets(h *histogram.Histogram) uint64 {
+	return countBuckets(len(h.PositiveBuckets), len(h.NegativeBuckets), h.ZeroCount > 0)
+}
+
+// countFloatHistogramBuckets returns the number of buckets in a float histogram.
+func countFloatHistogramBuckets(fh *histogram.FloatHistogram) uint64 {
+	return countBuckets(len(fh.PositiveBuckets), len(fh.NegativeBuckets), fh.ZeroCount > 0)
+}
```

Comment:
Is it the right place to have those functions? Should we move these helper functions to histogram package